runtime/lua/vim/_editor.lua

```diff
@@ -789,7 +790,9 @@ function vim.deprecate(name, alternative, version, plugin)
   plugin = plugin or 'Nvim'
   message = alternative and (message .. ', use ' .. alternative .. ' instead.') or message
   message = message .. ' See :h deprecated\nThis function will be removed in ' .. plugin .. ' version ' .. version
-  vim.notify_once(message, vim.log.levels.WARN)
+  if vim.notify_once(message, vim.log.levels.WARN) then
+    vim.notify(debug.traceback('', 2):sub(2), vim.log.levels.WARN)
```

Comment:
Looks like you can't just do `debug.traceback(nil, 2)` if you want to start the traceback from the function that calls `vim.deprecate`, and you have to include some message. Alternatively, it could just be simply `debug.traceback()` and the traceback could start from this line.